### PR TITLE
added employee models, controlles and reposotories

### PR DESCRIPTION
**Pull request by Bazhanel**

### CODE REVIEW

File: src/main/java/com/danit/models/eployee/Department.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  private Date dateFrom;
+
+  @Column(name = "dateto")
+  @JsonDeserialize(using = CustomDateAndTimeDeserialize.class)
+  @JsonFormat(
+      shape = JsonFormat.Shape.STRING,
+      pattern = "yyyy-MM-dd")
+  private Date dateTo;
+
+  @Column(name = "hier_level", nullable = false)
```
Comment (Thealoner) [inline src/main/java/com/danit/models/eployee/Department.java]: higher_level?

File: src/main/java/com/danit/controllers/employee/GroupTrainingController.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class GroupTrainingController {
+  @Autowired
+  private GroupTrainingRepository groupTrainingRepository;
+
+  @GetMapping("/grouptraining")
```
Comment (Thealoner) [inline src/main/java/com/danit/controllers/employee/GroupTrainingController.java]: We usually separate words within endpoint names with underscore. So I suggest to use '/group_training' here.

File: src/main/java/com/danit/controllers/employee/EmployeeController.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  @Autowired
+  private EmployeeRepository employeeRepository;
+
+  @GetMapping("/employee")
+  public List<Employee> retrieveAllEmployee() {
+    return employeeRepository.findAll();
+  }
+
+  @GetMapping("/employee/{id}")
+  public Employee retrieveStudent(@PathVariable long id) {
```
Comment (Thealoner) [inline src/main/java/com/danit/controllers/employee/EmployeeController.java]: retrieveEmployee?

File: src/main/java/com/danit/controllers/employee/EmployeeController.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    Optional<Employee> employee = employeeRepository.findById(id);
+
+    if (!employee.isPresent()) {
+      throw new EntityNotFoundException("id-" + id);
+    }
+    return employee.get();
+  }
+
+  @DeleteMapping("/employee/{id}")
+  public void deleteStudent(@PathVariable long id) {
```
Comment (Thealoner) [inline src/main/java/com/danit/controllers/employee/EmployeeController.java]: deleteEmployee?

File: src/main/java/com/danit/controllers/employee/EmployeeController.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}")
+        .buildAndExpand(savedEmployee.getId()).toUri();
+
+    return ResponseEntity.created(location).build();
+
+  }
+
+  @PutMapping("/employee/{id}")
+  public ResponseEntity<Object> updateStudent(@RequestBody Employee employee, @PathVariable long id) {
```
Comment (Thealoner) [inline src/main/java/com/danit/controllers/employee/EmployeeController.java]: updateEmployee?

File: src/main/java/com/danit/controllers/employee/GymController.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  @Autowired
+  private GymRepository gymRepository;
+
+  @GetMapping("/gym")
+  public List<Gym> retrieveAllGym() {
+    return gymRepository.findAll();
+  }
+
+  @GetMapping("/gym/{id}")
+  public Gym retrieveStudent(@PathVariable long id) {
```
[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/GymController.java]: Copy Paste method was used

File: src/main/java/com/danit/controllers/employee/TrainingTypeController.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  @Autowired
+  private TrainingTypeRepository trainingTypeRepository;
+
+  @GetMapping("/trainingtype")
+  public List<TrainingType> retrieveAllTrainingType() {
+    return trainingTypeRepository.findAll();
+  }
+
+  @GetMapping("/trainingType/{id}")
+  public TrainingType retrieveStudent(@PathVariable long id) {
```
[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/TrainingTypeController.java]: Copy Paste method was used

File: src/main/java/com/danit/controllers/employee/TrainingTypeController.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    Optional<TrainingType> trainingType = trainingTypeRepository.findById(id);
+
+    if (!trainingType.isPresent()) {
+      throw new EntityNotFoundException("id-" + id);
+    }
+    return trainingType.get();
+  }
+
+  @DeleteMapping("/trainingtype/{id}")
+  public void deleteStudent(@PathVariable long id) {
```
[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/TrainingTypeController.java]: Copy Paste method was used

File: src/main/java/com/danit/controllers/employee/GroupTrainingController.java (hunk 2 above)
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/GroupTrainingController.java]: ok

File: src/main/java/com/danit/controllers/employee/EmployeeController.java (hunk 3 above)
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/EmployeeController.java]: Copy Paste method was used

File: src/main/java/com/danit/controllers/employee/EmployeeController.java (hunk 4 above)
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/EmployeeController.java]: Copy Paste method was used

File: src/main/java/com/danit/controllers/employee/EmployeeController.java (hunk 5 above)
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/EmployeeController.java]: Copy Paste method was used

File: src/main/java/com/danit/models/eployee/Category.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Data
+@Table(name = "category")
+public class Category {
```
[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/models/eployee/Category.java]: ok

File: src/main/java/com/danit/models/eployee/Post.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Data
+@Entity
+@Table(name = "post")
+public class Post {
```
[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/models/eployee/Post.java]: ok

[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/models/eployee/Post.java]: will be position

File: src/main/java/com/danit/controllers/ServiceCategoryController.java [hunk 11]
```diff
@@ -55,19 +55,19 @@ void updateServiceCategories(@RequestBody List<ServiceCategory> serviceCategorie
 
   @PutMapping("/service_categories/{id}")
   void updateServiceCategory(@RequestBody ServiceCategory serviceCategory, Principal principal) {
-    logger.info(principal.getName() + " is updating service category data: " + serviceCategory);
+    logger.info(principal.getName() + " is updating service employeeCategory data: " + serviceCategory);
```
Comment (Thealoner) [inline src/main/java/com/danit/controllers/ServiceCategoryController.java]: Autoreplaced what was not supposed to be changed

File: src/main/java/com/danit/controllers/employee/EmployeeController.java [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class EmployeeController {
```
[the remark this answers is not in the document]
Comment (Thealoner) [reply src/main/java/com/danit/controllers/employee/EmployeeController.java]: We can do this later, after merging, if it's too much right now

File: src/main/java/com/danit/controllers/employee/CompanyController.java [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+public class CompanyController {
+  @Autowired
+  private CompanyRepository companyRepository;
+
+  @GetMapping("/company")
```
[the remark this answers is not in the document]
Comment (Thealoner) [reply src/main/java/com/danit/controllers/employee/CompanyController.java]: We can change this later

[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/CompanyController.java]: I'll do it later, after merge

[the remark this answers is not in the document]
Comment (Bazhanel) [reply src/main/java/com/danit/controllers/employee/CompanyController.java]: organization will be removed

**Comment by Bazhanel** [comment]
Thanks

**Pull request merged**